### PR TITLE
remove command fetcher

### LINKED ISSUE #5428: get rid of airbyte-command-fetcher pods
Airbyte on kubernetes starts pods named `airbyte-command-fetcher-*****`.

from [what I read](https://github.com/airbytehq/airbyte/blob/master/airbyte-workers/src/main/java/io/airbyte/workers/process/KubePodProcess.java#L145), they are only used to get the value of an env var (`AIRBYTE_ENTRYPOINT`)from the image, that is then only used as part of the startup script for the image itself.

you could instead use the variable itself in the startup script (to something like `eval $AIRBYTE_ENTRYPOINT write --config xxx --catalog yyy < stdin.pipe > stdout.pipe 2> stderr.pipe` + rest of current boilerplate) and completely avoid starting a pod, which is both slowing down the UI, and heavy because there is a single setting for all worker resources.

### PR DESCRIPTION
**Pull request by jrhizor**
resolves https://github.com/airbytehq/airbyte/issues/5428

The previous flow was:
* create a command fetcher pod
* read the output from that pod 
* use the output to create a new pod that actually runs

The new flow is:
* create the pod and have the main container determine if the env variable is set properly or not and start!

This saves a few seconds on every UI operation on Kubernetes! Great idea from @m-ronchi! 

The remaining changes in this PR allow `KubePodProcessIntegrationTest` to run locally on docker-desktop by using `host.docker.internal` as the local ip/host in the tests.

### CODE REVIEW

**Comment by jrhizor** [comment]
Ready for an initial review. Still need to make sure that we properly escape any entrypoint override string we're passing in.

File: airbyte-workers/src/main/java/io/airbyte/workers/process/KubePodProcess.java [hunk 1]
```diff
@@ -120,6 +118,7 @@
   private static final int STDIN_REMOTE_PORT = 9001;
   private static final Map<String, String> AIRBYTE_POD_LABELS = Map.of("airbyte", "worker-pod");
 
+  private final String processRunnerHost;
```
Comment (davinchia) [inline airbyte-workers/src/main/java/io/airbyte/workers/process/KubePodProcess.java:121]: looks like we don't need to store this as class state and having it within the function is sufficient.

File: airbyte-workers/src/main/java/io/airbyte/workers/process/KubeProcessFactory.java [hunk 2]
```diff
@@ -47,23 +49,37 @@
...
    * @param namespace kubernetes namespace where spawned pods will live
    * @param officialClient official kubernetes client
    * @param fabricClient fabric8 kubernetes client
    * @param kubeHeartbeatUrl a url where if the response is not 200 the spawned process will fail
    *        itself
-   * @param workerPorts a set of ports that can be used for IO socket servers
+   * @param processRunnerHost is the local host or ip of the machine running the process factory.
+   *        injectable for testing.
    */
   public KubeProcessFactory(String namespace,
```
Comment (davinchia) [inline airbyte-workers/src/main/java/io/airbyte/workers/process/KubeProcessFactory.java:73]: nit: VisibleForTesting annotation?

File: airbyte-workers/src/main/java/io/airbyte/workers/process/KubePodProcess.java [hunk 3]
```diff
@@ -345,7 +305,7 @@ public KubePodProcess(ApiClient officialClient,
     Container main = getMain(
         image,
         usesStdin,
-        entrypoint,
+        entrypointOverride,
```
Comment (davinchia) [inline airbyte-workers/src/main/java/io/airbyte/workers/process/KubePodProcess.java:308]: now that we no longer need to specify entrypoint - what is the override for? looks like mostly tests?

File: airbyte-workers/src/main/java/io/airbyte/workers/process/KubePodProcess.java [hunk 4]
```diff
@@ -120,6 +118,7 @@
   private static final int STDIN_REMOTE_PORT = 9001;
   private static final Map<String, String> AIRBYTE_POD_LABELS = Map.of("airbyte", "worker-pod");
```
Comment (davinchia) [inline airbyte-workers/src/main/java/io/airbyte/workers/process/KubePodProcess.java:119]: we forgot to do this when we added the command fetcher pod, can we add a note in the class docstring about the assumption that images need an AIRBYTE_ENTRYPOINT env var set?

**Review by davinchia — APPROVED** [review]
Looks good!

File: airbyte-workers/src/main/java/io/airbyte/workers/process/KubePodProcess.java (hunk 3 above)
Comment (jrhizor) [reply airbyte-workers/src/main/java/io/airbyte/workers/process/KubePodProcess.java:308]: This is used by custom operators.

**Comment by jrhizor** [comment]
@davinchia PTAL. I addressed comments and added escaping. 

@cgardens and @davinchia I would appreciate a close look at escaping to see if this approach is reasonable.

**Review by cgardens — APPROVED** [review]
🎉 ❤️  awesome. Thank @m-ronchi for calling this out.

**Comment by cgardens** [comment]
and yes, the approach to escaping makes sense to me.

**Comment by michel-tricot** [comment]
Super nice!

**Review by davinchia — APPROVED** [review]
Nice!

**Pull request merged**